Content.Server/Chat/V2/Repository/ChatRepository.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System.Linq;
+using System.Numerics;
+using Content.Shared.Chat.V2;
+using Content.Shared.Chat.V2.Repository;
+using Robust.Server.Player;
+using Robust.Shared.Replays;
+
+namespace Content.Server.Chat.V2.Repository;
+
+/// <summary>
+/// Stores <see cref="IChatEvent"/>, gives them UIDs, and issues <see cref="ChatEventCreated"/>.
+/// </summary>
+/// <remarks>
+/// This is an <see cref="EntitySystem"/> because:
+/// <list type="number"><item>It raises events</item>
+/// <item>It needs to extract user UIDs from entities</item>
+/// <item>Making this be a manager means more auto-wiring boilerplate</item></list>
+/// </remarks>
+public sealed class ChatRepository : EntitySystem
+{
+    [Dependency] private readonly IReplayRecordingManager _replay = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+
+    // Clocks should start at 1, as 0 indicates "clock not set" or "clock forgotten to be set by bad programmer".
+    private uint _clock = 1;
```

Comment:
Can null be used instead of 0? Also I suggest giving it a more descriptive name like _nextMessageId.

Reply:
uint is a primitive, i.e. no nulls.